csharp/src/Ice/InputStream.cs

```diff
@@ -35,15 +35,6 @@ public sealed class InputStream
         /// <value>The communicator.</value>
         public Communicator Communicator { get; }
 
-        /// <summary>
-        /// The encoding used when reading from this stream.
-        /// </summary>
-        /// <value>The encoding.</value>
-        public Encoding Encoding { get; private set; }
```

Comment:
I don't understand how we could get rid of Encoding in either InputStream or OutputStream. Both InputStream and OutputStream need to know the encoding, and this encoding can change with the nested endpoint encaps (which can have a different encoding than the parent's encaps).
Right now we've not added the 2.0 encoding yet and as a result there is only one encoding throughout the code base (1.1), but that's a temporary situation.

Reply:
We still keep the encoding but it is a private member, isn't this enough here?
https://github.com/zeroc-ice/ice/blob/c3c1273167be944a231628092fa7084b54ebbae4/csharp/src/Ice/InputStream.cs#L110

Reply:
I was assuming that the encoding parameter was removed from the InputStream constructor because we assumed all the data read outside an encapsulation was always 1.1 encoded. 

If the ice2 protocol requires another encoding, we might need to add it back but perhaps for now it's best to get rid of it and then see how we'll deal with ice2?

Reply:
With the ice2 protocol, the header will be encoded with the 2.0 encoding.
Also note that the `Encoding` property of the various frames corresponds to the payload encoding, not the header encoding (and yes right now it's all 1.1 so we can't easily tell them apart).